src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.http.api;
+
+import static com.google.common.collect.ImmutableMap.of;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.collection.IsMapContaining.hasEntry;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mule.extension.http.api.HttpHeaders.Names.AUTHORIZATION;
+
+import org.mule.apache.xerces.impl.dv.util.Base64;
+import org.mule.extension.http.api.HttpRequestAttributes;
+import org.mule.extension.http.api.listener.HttpBasicAuthenticationFilter;
+import org.mule.runtime.api.security.Authentication;
+import org.mule.runtime.api.security.CredentialsBuilder;
+import org.mule.runtime.api.security.DefaultMuleAuthentication;
+import org.mule.runtime.api.util.MultiMap;
+import org.mule.runtime.core.api.security.DefaultMuleCredentials;
+import org.mule.runtime.extension.api.security.AuthenticationHandler;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpListenerAuthenticationHandlerTestCase extends AbstractHttpAttributesTestCase {
```

Comment:
Was there no way to test this with an integration test?

Reply:
It was our first approach, but there was no easy way to set the authentication beforehand.